### PR TITLE
Atmel RF driver corrections

### PR DESCRIPTION
**Pull request by mikaleppanen**

### CODE REVIEW

**Comment by mikaleppanen** [comment]
@kjbracey-arm @JarkkoPaso @SeppoTakalo Please review.

File: source/driverRFPhy.c [hunk 1]
```diff
@@ -476,7 +473,7 @@ void rf_off(void)
         }
         rf_if_change_trx_state(TRX_OFF);
         rf_if_enable_slptr();
-        rf_flags_clear(~RFF_ON);
+        rf_flags_clear(RFF_ON);
```
Comment (kjbracey-arm) [inline source/driverRFPhy.c]: This used to clear "everything except ON", now it clears only ON.

I think logically this function should be explicitly clearing ON|TX|RX|CCA - it is shutting the chip down (rf_flags_reset?) Then callers don't need to do that.

File: source/driverRFPhy.c [hunk 2]
```diff
@@ -625,7 +624,7 @@ void rf_receive(void)
         }
         //TODO: rf_if_delay_function(50);
         /*Wake up from sleep state*/
-        if(rf_if_read_trx_state() == 0x00 || rf_if_read_trx_state() == 0x1f)
+        if(sleep_enabled)
```
Comment (kjbracey-arm) [inline source/driverRFPhy.c]: What do we think this code does? Seems pointless?

**Comment by mikaleppanen** [comment]
Updated based on comments

File: source/driverRFPhy.c (hunk 1 above)
Comment (mikaleppanen) [inline source/driverRFPhy.c]: Added that

File: source/driverRFPhy.c (hunk 2 above)
Comment (mikaleppanen) [inline source/driverRFPhy.c]: Moved sleep disable to rf_on

**Pull request merged**